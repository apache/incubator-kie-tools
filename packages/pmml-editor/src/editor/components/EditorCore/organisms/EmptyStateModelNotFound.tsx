/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from "react";
import { EmptyState, EmptyStateBody, EmptyStateIcon, EmptyStateVariant, Title } from "@patternfly/react-core";
import { ExclamationTriangleIcon } from "@patternfly/react-icons";

export const EmptyStateModelNotFound = () => (
  <EmptyState data-testid="empty-state-model-not-found" variant={EmptyStateVariant.small}>
    <EmptyStateIcon icon={ExclamationTriangleIcon} />
    <Title headingLevel="h4" size="lg">
      Not found.
    </Title>
    <EmptyStateBody>Something went wrong. An attempt has been made to view a Model that does not exist.</EmptyStateBody>
  </EmptyState>
);
