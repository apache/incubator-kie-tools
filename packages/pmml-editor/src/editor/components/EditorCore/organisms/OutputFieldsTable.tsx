/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from "react";
import { useEffect, useRef } from "react";
import { Bullseye, Form } from "@patternfly/react-core";
import { OutputField } from "@kogito-tooling/pmml-editor-marshaller";
import { OutputFieldEditRow, OutputFieldRow } from "../molecules";
import "./OutputFieldsTable.scss";
import { Operation } from "../../EditorScorecard";
import { EmptyStateNoOutput } from "./EmptyStateNoOutput";

interface OutputFieldsTableProps {
  activeOperation: Operation;
  onAddOutputField: () => void;
  onEditOutputField: (index: number) => void;
  onDeleteOutputField: (index: number) => void;
  activeOutputFieldIndex: number | undefined;
  activeOutputField: OutputField;
  setActiveOutputField: (_output: OutputField) => void;
  outputs: OutputField[];
  validateOutputFieldName: (index: number | undefined, name: string | undefined) => boolean;
  viewExtendedProperties: () => void;
  onCommit: () => void;
  onCancel: () => void;
}

export const OutputFieldsTable = (props: OutputFieldsTableProps) => {
  const {
    activeOperation,
    onAddOutputField,
    onEditOutputField,
    onDeleteOutputField,
    activeOutputFieldIndex,
    activeOutputField,
    setActiveOutputField,
    outputs,
    validateOutputFieldName,
    viewExtendedProperties,
    onCommit,
    onCancel
  } = props;

  const addOutputRowRef = useRef<HTMLDivElement | null>(null);

  useEffect(() => {
    if (activeOperation === Operation.CREATE_OUTPUT && addOutputRowRef.current) {
      addOutputRowRef.current.scrollIntoView({ behavior: "smooth" });
    }
  }, [activeOperation]);

  const onDelete = (index: number | undefined) => {
    if (index !== undefined) {
      onDeleteOutputField(index);
    }
  };

  const onValidateOutputFieldName = (index: number | undefined, name: string | undefined): boolean => {
    return validateOutputFieldName(index, name);
  };

  return (
    <Form>
      <section>
        {outputs.map((o, index) => {
          if (activeOutputFieldIndex === index) {
            return (
              <OutputFieldEditRow
                key={index}
                activeOutputFieldIndex={index}
                activeOutputField={activeOutputField}
                setActiveOutputField={setActiveOutputField}
                validateOutputName={_name => onValidateOutputFieldName(index, _name)}
                viewExtendedProperties={viewExtendedProperties}
                onCommit={onCommit}
                onCancel={onCancel}
              />
            );
          } else {
            return (
              <OutputFieldRow
                key={index}
                activeOutputFieldIndex={index}
                activeOutputField={o}
                onEditOutputField={() => onEditOutputField(index)}
                onDeleteOutputField={() => onDelete(index)}
                isDisabled={
                  !(activeOutputFieldIndex === undefined || activeOutputFieldIndex === index) ||
                  activeOperation !== Operation.NONE
                }
              />
            );
          }
        })}
        {activeOperation === Operation.CREATE_OUTPUT && (
          <div key={undefined} ref={addOutputRowRef}>
            <OutputFieldEditRow
              key={"add"}
              activeOutputFieldIndex={undefined}
              activeOutputField={activeOutputField}
              setActiveOutputField={setActiveOutputField}
              validateOutputName={_name => onValidateOutputFieldName(undefined, _name)}
              viewExtendedProperties={viewExtendedProperties}
              onCommit={onCommit}
              onCancel={onCancel}
            />
          </div>
        )}
      </section>
      {outputs.length === 0 && activeOperation !== Operation.CREATE_OUTPUT && (
        <Bullseye>
          <EmptyStateNoOutput onAddOutputField={onAddOutputField} />
        </Bullseye>
      )}
    </Form>
  );
};
