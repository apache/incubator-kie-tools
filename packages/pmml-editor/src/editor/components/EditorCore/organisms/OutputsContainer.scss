/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.outputs-container {
  height: 60vh;

  &__list__header {
    margin-left: 16px;
  }

  &__list__container {
    height: calc(100% - 68px);
  }

  &__list {
    height: calc(100% - 64px);
    overflow-y: auto;
    padding: 1em;
    background-color: var(--pf-global--BackgroundColor--200);
  }
}
