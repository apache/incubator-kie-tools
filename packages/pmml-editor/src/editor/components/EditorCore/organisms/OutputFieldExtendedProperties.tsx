/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { Form, FormGroup, TextInput } from "@patternfly/react-core";
import "../organisms/OutputFieldsTable.scss";
import { FieldName, OpType, OutputField, RankOrder, ResultFeature } from "@kogito-tooling/pmml-editor-marshaller";
import { GenericSelector } from "../../EditorScorecard/atoms";

interface OutputFieldExtendedPropertiesProps {
  activeOutputField: OutputField;
  setActiveOutputField: (_output: OutputField) => void;
}

const GenericSelectorEditor = (
  id: string,
  items: string[],
  selection: string,
  onSelect: (_selection: string) => void
) => {
  return <GenericSelector id={id} items={items} selection={selection} onSelect={onSelect} />;
};

export const OutputFieldExtendedProperties = (props: OutputFieldExtendedPropertiesProps) => {
  const { activeOutputField, setActiveOutputField } = props;

  const toNumber = (value: string): number | undefined => {
    if (value === "") {
      return undefined;
    }
    const n = Number(value);
    if (isNaN(n)) {
      return undefined;
    }
    return n;
  };

  const optypeEditor = GenericSelectorEditor(
    "output-optype",
    ["", "categorical", "continuous", "ordinal"],
    (activeOutputField.optype ?? "").toString(),
    _selection =>
      setActiveOutputField({
        ...activeOutputField,
        optype: _selection === "" ? undefined : (_selection as OpType)
      })
  );

  const featureEditor = GenericSelectorEditor(
    "output-feature",
    [
      "",
      "affinity",
      "antecedent",
      "clusterAffinity",
      "clusterId",
      "confidence",
      "consequent",
      "decision",
      "entityAffinity",
      "entityId",
      "leverage",
      "lift",
      "predictedDisplayValue",
      "predictedValue",
      "probability",
      "reasonCode",
      "residual",
      "rule",
      "ruleId",
      "ruleValue",
      "standardDeviation",
      "standardError",
      "support",
      "transformedValue",
      "warning"
    ],
    (activeOutputField.feature ?? "").toString(),
    _selection =>
      setActiveOutputField({
        ...activeOutputField,
        feature: _selection === "" ? undefined : (_selection as ResultFeature)
      })
  );

  const rankOrderEditor = GenericSelectorEditor(
    "output-rankOrder",
    ["", "ascending", "descending"],
    (activeOutputField.rankOrder ?? "").toString(),
    _selection =>
      setActiveOutputField({
        ...activeOutputField,
        rankOrder: _selection === "" ? undefined : (_selection as RankOrder)
      })
  );

  const isFinalResultEditor = GenericSelectorEditor(
    "output-isFinalResult",
    ["", "true", "false"],
    (activeOutputField.isFinalResult ?? "").toString(),
    _selection =>
      setActiveOutputField({
        ...activeOutputField,
        isFinalResult: _selection === "" ? undefined : Boolean(_selection)
      })
  );

  return (
    <Form>
      <FormGroup
        label="optype"
        fieldId="output-optype-helper"
        helperText="Indicates the admissible operations on the values."
      >
        {optypeEditor}
      </FormGroup>
      <FormGroup
        label="Target field"
        fieldId="output-targetField-helper"
        helperText="Target field for the Output field."
      >
        <TextInput
          type="text"
          id="output-targetField"
          name="output-targetField"
          aria-describedby="output-targetField-helper"
          value={(activeOutputField.targetField ?? "").toString()}
          onChange={e => setActiveOutputField({ ...activeOutputField, targetField: e as FieldName })}
        />
      </FormGroup>
      <FormGroup
        label="Feature"
        fieldId="output-feature-helper"
        helperText="Specifies the value the output field takes from the computed mining result."
      >
        {featureEditor}
      </FormGroup>
      <FormGroup
        label="Value"
        fieldId="output-value-helper"
        helperText="Used in conjunction with result features referring to specific values."
      >
        <TextInput
          type="text"
          id="output-value"
          name="output-value"
          aria-describedby="output-value-helper"
          value={(activeOutputField.value ?? "").toString()}
          onChange={e => setActiveOutputField({ ...activeOutputField, value: e })}
        />
      </FormGroup>
      <FormGroup
        label="Rank"
        fieldId="output-rank-helper"
        helperText="Specifies the rank of the feature value from the mining result that should be selected."
      >
        <TextInput
          type="number"
          id="output-rank"
          name="output-rank"
          aria-describedby="output-rank-helper"
          value={activeOutputField.rank}
          onChange={e => setActiveOutputField({ ...activeOutputField, rank: toNumber(e) })}
        />
      </FormGroup>
      <FormGroup
        label="Rank order"
        fieldId="output-rankOrder-helper"
        helperText="Determines the sorting order when ranking the results."
      >
        {rankOrderEditor}
      </FormGroup>
      <FormGroup
        label="Segment Id"
        fieldId="output-segmentId-helper"
        helperText="Provides an approach to deliver results from Segments."
      >
        <TextInput
          type="text"
          id="output-segmentId"
          name="output-segmentId"
          aria-describedby="output-segmentId-helper"
          value={activeOutputField.segmentId}
          onChange={e => setActiveOutputField({ ...activeOutputField, segmentId: e })}
        />
      </FormGroup>
      <FormGroup
        label="Final result?"
        fieldId="output-isFinalResult-helper"
        helperText="Should the field be returned to the user or is only used as input."
      >
        {isFinalResultEditor}
      </FormGroup>
    </Form>
  );
};
