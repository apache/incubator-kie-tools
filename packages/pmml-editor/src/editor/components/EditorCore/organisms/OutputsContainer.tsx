/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from "react";
import { Output, OutputField } from "@kogito-tooling/pmml-editor-marshaller";
import { EmptyStateNoOutput } from "./EmptyStateNoOutput";
import { Bullseye, Button, Flex, FlexItem } from "@patternfly/react-core";
import { PlusIcon } from "@patternfly/react-icons";
import { OutputsTable } from "./OutputsTable";
import { Operation } from "../../EditorScorecard";
import "./OutputsContainer.scss";

interface OutputsContainerProps {
  modelIndex: number;
  activeOperation: Operation;
  setActiveOperation: (operation: Operation) => void;
  output?: Output;
  validateOutputName: (index: number | undefined, name: string | undefined) => boolean;
}

export const OutputsContainer = (props: OutputsContainerProps) => {
  const { activeOperation, setActiveOperation, output, validateOutputName } = props;

  const addOutput = () => {
    setActiveOperation(Operation.CREATE_OUTPUT);
  };

  return (
    <div className="outputs-container">
      <Flex style={{ margin: "1em 0 2em 0" }}>
        <FlexItem>
          <Button
            variant="secondary"
            onClick={e => addOutput()}
            isDisabled={activeOperation !== Operation.NONE}
            icon={<PlusIcon />}
            iconPosition="left"
          >
            Add Output
          </Button>
        </FlexItem>
      </Flex>
      {(output?.OutputField ?? []).length > 0 && (
        <div className="outputs-container__list">
          <div className="outputs-container__list--container">
            <OutputsTable
              activeOperation={activeOperation}
              setActiveOperation={setActiveOperation}
              validateOutputName={validateOutputName}
              deleteOutput={_index => null}
              commit={(index, text) => null}
              outputs={output?.OutputField as OutputField[]}
            />
          </div>
        </div>
      )}
      {(output?.OutputField ?? []).length === 0 && (
        <Bullseye>
          <EmptyStateNoOutput addOutput={addOutput} />
        </Bullseye>
      )}
    </div>
  );
};
