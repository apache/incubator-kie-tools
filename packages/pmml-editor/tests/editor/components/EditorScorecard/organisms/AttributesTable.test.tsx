/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { fireEvent, render } from "@testing-library/react";
import * as React from "react";
import { Attribute, Characteristic, PMML, Scorecard } from "@kie-tools/pmml-editor-marshaller";
import { Operation, OperationContext } from "../../../../../dist/editor/components/EditorScorecard";
import { AttributesTable } from "../../../../../src/editor/components/EditorScorecard/organisms";
import { Provider } from "react-redux";
import { createStore, Store } from "redux";
import { AllActions } from "../../../../../src/editor/reducers";

const viewAttribute = jest.fn((index) => {});
const deleteAttribute = jest.fn((index) => {});
const onCommit = jest.fn((index, partial) => {});

let attributes: Attribute[];
let characteristics: Characteristic[];
let pmml: PMML;
let scorecard: Scorecard;
let store: Store<PMML, AllActions>;

beforeEach(() => {
  viewAttribute.mockReset();
  deleteAttribute.mockReset();
  onCommit.mockReset();

  attributes = [{ reasonCode: "A1" }, { reasonCode: "A2" }];

  characteristics = [
    {
      name: "characteristic1",
      Attribute: attributes,
    },
  ];

  scorecard = {
    MiningSchema: { MiningField: [] },
    Characteristics: { Characteristic: characteristics },
    functionName: "regression",
  };
  pmml = { Header: {}, DataDictionary: { DataField: [] }, models: [scorecard], version: "1" };
  store = createStore((state) => state, pmml);
});

describe("AttributesTable", () => {
  test("Attribute:Render", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <OperationContext.Provider
          value={{
            activeOperation: Operation.UPDATE_CHARACTERISTIC,
            setActiveOperation: (operation) => {},
          }}
        >
          <AttributesTable
            modelIndex={0}
            characteristicIndex={0}
            characteristic={characteristics[0]}
            areReasonCodesUsed={true}
            viewAttribute={viewAttribute}
            deleteAttribute={deleteAttribute}
            onCommit={onCommit}
          />
        </OperationContext.Provider>
      </Provider>
    );
    const container = getByTestId("attributes-table");
    expect(container).toMatchSnapshot();

    expect(getByTestId("attribute-n0")).not.toBeUndefined();
    expect(getByTestId("attribute-n1")).not.toBeUndefined();
  });

  test("Attribute:DeleteWithIconClick", () => {
    const deleteAttributeImpl = jest.fn((index) => {
      scorecard.Characteristics.Characteristic[0].Attribute =
        scorecard.Characteristics.Characteristic[0].Attribute.slice(index, 1);
    });

    const { getByTestId, queryByTestId, rerender } = render(
      <Provider store={store}>
        <OperationContext.Provider
          value={{
            activeOperation: Operation.UPDATE_CHARACTERISTIC,
            setActiveOperation: (operation) => {},
          }}
        >
          <AttributesTable
            modelIndex={0}
            characteristicIndex={0}
            characteristic={characteristics[0]}
            areReasonCodesUsed={true}
            viewAttribute={viewAttribute}
            deleteAttribute={deleteAttributeImpl}
            onCommit={onCommit}
          />
        </OperationContext.Provider>
      </Provider>
    );

    //Get the first Attribute and focus the row that will make the delete icon visible
    const characteristic0 = getByTestId("attribute-n0");
    expect(characteristic0).not.toBeUndefined();
    fireEvent.focus(characteristic0, {});

    rerender(
      <Provider store={store}>
        <OperationContext.Provider
          value={{
            activeOperation: Operation.UPDATE_CHARACTERISTIC,
            setActiveOperation: (operation) => {},
          }}
        >
          <AttributesTable
            modelIndex={0}
            characteristicIndex={0}
            characteristic={characteristics[0]}
            areReasonCodesUsed={true}
            viewAttribute={viewAttribute}
            deleteAttribute={deleteAttributeImpl}
            onCommit={onCommit}
          />
        </OperationContext.Provider>
      </Provider>
    );

    const characteristic0deleteIcon = getByTestId("attribute-n0__delete");
    expect(characteristic0deleteIcon).not.toBeUndefined();
    fireEvent.click(characteristic0deleteIcon, {});

    expect(deleteAttributeImpl).toBeCalledWith(0);

    rerender(
      <Provider store={store}>
        <OperationContext.Provider
          value={{
            activeOperation: Operation.UPDATE_CHARACTERISTIC,
            setActiveOperation: (operation) => {},
          }}
        >
          <AttributesTable
            modelIndex={0}
            characteristicIndex={0}
            characteristic={characteristics[0]}
            areReasonCodesUsed={true}
            viewAttribute={viewAttribute}
            deleteAttribute={deleteAttributeImpl}
            onCommit={onCommit}
          />
        </OperationContext.Provider>
      </Provider>
    );

    expect(getByTestId("attribute-n0")).not.toBeUndefined();
    expect(queryByTestId("attribute-n1")).toBeNull();
  });

  test("Attribute:DeleteWithIconKeyDown", () => {
    const deleteAttributeImpl = jest.fn((index) => {
      scorecard.Characteristics.Characteristic[0].Attribute =
        scorecard.Characteristics.Characteristic[0].Attribute.slice(index, 1);
    });

    const { getByTestId, queryByTestId, rerender } = render(
      <Provider store={store}>
        <OperationContext.Provider
          value={{
            activeOperation: Operation.UPDATE_CHARACTERISTIC,
            setActiveOperation: (operation) => {},
          }}
        >
          <AttributesTable
            modelIndex={0}
            characteristicIndex={0}
            characteristic={characteristics[0]}
            areReasonCodesUsed={true}
            viewAttribute={viewAttribute}
            deleteAttribute={deleteAttributeImpl}
            onCommit={onCommit}
          />
        </OperationContext.Provider>
      </Provider>
    );

    //Get the first Attribute and focus the row that will make the delete icon visible
    const characteristic0 = getByTestId("attribute-n0");
    expect(characteristic0).not.toBeUndefined();
    fireEvent.focus(characteristic0, {});

    rerender(
      <Provider store={store}>
        <OperationContext.Provider
          value={{
            activeOperation: Operation.UPDATE_CHARACTERISTIC,
            setActiveOperation: (operation) => {},
          }}
        >
          <AttributesTable
            modelIndex={0}
            characteristicIndex={0}
            characteristic={characteristics[0]}
            areReasonCodesUsed={true}
            viewAttribute={viewAttribute}
            deleteAttribute={deleteAttributeImpl}
            onCommit={onCommit}
          />
        </OperationContext.Provider>
      </Provider>
    );

    const characteristic0deleteIcon = getByTestId("attribute-n0__delete");
    expect(characteristic0deleteIcon).not.toBeUndefined();
    fireEvent.keyDown(characteristic0deleteIcon, { key: "Enter" });

    expect(deleteAttributeImpl).toBeCalledWith(0);

    rerender(
      <Provider store={store}>
        <OperationContext.Provider
          value={{
            activeOperation: Operation.UPDATE_CHARACTERISTIC,
            setActiveOperation: (operation) => {},
          }}
        >
          <AttributesTable
            modelIndex={0}
            characteristicIndex={0}
            characteristic={characteristics[0]}
            areReasonCodesUsed={true}
            viewAttribute={viewAttribute}
            deleteAttribute={deleteAttributeImpl}
            onCommit={onCommit}
          />
        </OperationContext.Provider>
      </Provider>
    );

    const characteristic0Rendered = getByTestId("attribute-n0");
    expect(characteristic0Rendered).not.toBeUndefined();
    expect(document.activeElement).toBe(characteristic0Rendered);
    expect(queryByTestId("attribute-n1")).toBeNull();
  });
});
