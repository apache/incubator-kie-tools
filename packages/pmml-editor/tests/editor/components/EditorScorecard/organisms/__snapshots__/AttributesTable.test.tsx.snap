// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributesTable Attribute:Render 1`] = `
<section
  data-testid="attributes-table"
>
  <article
    class="editable-item attribute-item-n0"
  >
    <article
      class="attribute-item"
      data-ouia-component-type="attribute-item"
      data-testid="attribute-n0"
      id="attribute-n0"
      tabindex="0"
    >
      <div
        class="pf-l-split pf-m-gutter"
        style="height: 100%;"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            class="pf-c-label pf-m-blue"
            tabindex="0"
          >
            <span
              class="pf-c-label__content"
            >
              <pre />
            </span>
            <button
              aria-disabled="false"
              aria-label="Close [object Object]"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
          </span>
        </div>
        <div
          class="pf-l-split__item pf-m-fill"
        >
          <span
            class="pf-c-label pf-m-cyan attribute-list__item__label"
          >
            <span
              class="pf-c-label__content"
            >
              <strong>
                Reason code
                :
              </strong>
               
              <span>
                A1
              </span>
            </span>
            <button
              aria-disabled="false"
              aria-label="Close [object Object], ,[object Object]"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <div
            class="pf-l-flex pf-m-align-items-center"
            style="height: 100%;"
          >
            <div
              class=""
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-plain attribute-item__delete"
                data-ouia-component-id="delete-attribute"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="attribute-n0__delete"
                id="attribute-n0__delete"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 448 512"
                  width="1em"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </article>
  </article>
  <article
    class="editable-item attribute-item-n1"
  >
    <article
      class="attribute-item"
      data-ouia-component-type="attribute-item"
      data-testid="attribute-n1"
      id="attribute-n1"
      tabindex="0"
    >
      <div
        class="pf-l-split pf-m-gutter"
        style="height: 100%;"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            class="pf-c-label pf-m-blue"
            tabindex="0"
          >
            <span
              class="pf-c-label__content"
            >
              <pre />
            </span>
            <button
              aria-disabled="false"
              aria-label="Close [object Object]"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-3"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
          </span>
        </div>
        <div
          class="pf-l-split__item pf-m-fill"
        >
          <span
            class="pf-c-label pf-m-cyan attribute-list__item__label"
          >
            <span
              class="pf-c-label__content"
            >
              <strong>
                Reason code
                :
              </strong>
               
              <span>
                A2
              </span>
            </span>
            <button
              aria-disabled="false"
              aria-label="Close [object Object], ,[object Object]"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-4"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <div
            class="pf-l-flex pf-m-align-items-center"
            style="height: 100%;"
          >
            <div
              class=""
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-plain attribute-item__delete"
                data-ouia-component-id="delete-attribute"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="attribute-n1__delete"
                id="attribute-n1__delete"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 448 512"
                  width="1em"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </article>
  </article>
</section>
`;
